src/lsstseries/analysis/structure_function/light_curve.py

```diff
@@ -0,0 +1,131 @@
+from typing import Optional
+import numpy as np
+
+MIN_OBSERVATIONS_REQUIRED_FOR_SF = 2
+
+
+class StructureFunctionLightCurve:
```

Comment:
Do we envision other analysis functions needing to create a LightCurve object? A lot of this specific code is handling the structurefunction case, but in the event that say, periodograms, needed a PeriodogramLightCurve class, it possibly could make sense to have a LightCurve baseclass established? Not sure

Reply:
That is a fair a point. Initially I was envisioning this as just being used by StructureFunction. However, it wouldn't be too hard to abstract this up one level to create a generic Lightcurve base class for that StructureFunctionLightCurve would inherit from. 

I could see the base class looking something like:

```class AnalysisLightCurve:  # name TBD
    def __init__(self, times, fluxes, errors):
        self._times = times
        self._fluxes = fluxes
        self._errors = errors

        self._process_input_data()
   
    def _process_input_data(self):
        # implement each of this in this class as well
        self._check_input_data_size_is_equal()
        self._filter_nans()
        self._check_input_data_length_is_sufficient()
```

Reply:
I created Issue #122 to track this work.